### ISSUE TITLE
APT DB needs to be updated before installing packages

### ISSUE THREAD

**Issue by pauldthomson** [abiosoft/colima#896]
### Description

Duplicate of #873 (recreated using the template)

I started getting an issue after a fresh install/VM start:
```shell
❯ colima start --vz-rosetta -t vz --mount-inotify
INFO[0000] starting colima
INFO[0000] runtime: docker
INFO[0000] creating and starting ...                     context=vm
> dpkg: error: cannot access archive '/Users/pthomson/Library/Caches/colima/packages/mantic/aarch64/*.deb': No such file or directory
FATA[0070] error starting vm: error at 'creating and starting': exit status 2
```
Digging in to it, it seems the issue is this line:
https://github.com/abiosoft/colima/blob/565ddd0c15a3068d0f11cb301691e3540f9b5596/environment/vm/lima/dependencies.go#L45
is getting no results:
```shell
pthomson@colima:~$ sudo apt-get install --reinstall --print-uris -qq docker.io | cut -d"'" -f2
E: Unable to locate package docker.io
E: Couldn't find any package by glob 'docker.io'
E: Couldn't find any package by regex 'docker.io'
```
seemingly due to apt's database needing to be updated. If I manually run `apt-get update`, I get:
```shell
pthomson@colima:~$ sudo apt-get install --reinstall --print-uris -qq htop | cut -d"'" -f2
http://ports.ubuntu.com/ubuntu-ports/pool/main/s/sgml-base/sgml-base_1.31_all.deb
http://ports.ubuntu.com/ubuntu-ports/pool/main/libn/libnl3/libnl-3-200_3.7.0-0.2_arm64.deb
http://ports.ubuntu.com/ubuntu-ports/pool/main/libn/libnl3/libnl-genl-3-200_3.7.0-0.2_arm64.deb
http://ports.ubuntu.com/ubuntu-ports/pool/main/h/htop/htop_3.2.2-2_arm64.deb
```
but the above line also never returns an error due to the pipe to `cut`. 

I'm not sure if `cloud-init` should be performing the `apt` update? Also not sure why it's just started but I guess the underlying cause might be in https://github.com/lima-vm/lima?

### Version

```shell
❯ limactl --version && qemu-img --version && colima version
limactl version 0.18.0
qemu-img version 8.1.2
Copyright (c) 2003-2023 Fabrice Bellard and the QEMU Project developers
colima version 0.6.5
git commit: a3adebfcd9da8c85614d81030033d02f04908a79
```
(sorry for the order, `colima version` doesn't terminate)

### Operating System

- [ ] macOS Intel <= 13 (Ventura)
- [ ] macOS Intel >= 14 (Sonoma)
- [ ] Apple Silicon <= 13 (Ventura)
- [X] Apple Silicon >= 14 (Sonoma)
- [ ] Linux

### Output of `colima status`

N/A (VM not started)

### Reproduction Steps

1. Remove all prior VMs and caches
2. Run `colima start --vz-rosetta -t vz --mount-inotify`
3. See error above

### Expected behaviour

Required packages are installed

**Comment by abiosoft** [comment]
Yeah, it can be modified to run `apt-get update`.
Also, the steps can be split to avoid a false `success` due to piping to `cut`.

Lastly, there is indeed a bug as it is supposed to fall back to regular `apt-get install` if that step fails.

**Comment by pauldthomson** [comment]
Looks like this is fixed in #884
